src/test/denialofservice_tests.cpp

```diff
@@ -34,6 +36,57 @@ static CService ip(uint32_t i)
 
 BOOST_FIXTURE_TEST_SUITE(denialofservice_tests, TestingSetup)
 
+void ConnectNode(ConnmanTestMsg& connman, PeerManager& peerman, CNode& node, ServiceFlags their_services) noexcept
+{
+    // Send version message
+    const CNetMsgMaker mm{0};
+    CSerializedNetMsg msg_version{
+        mm.Make(NetMsgType::VERSION,
+                PROTOCOL_VERSION,                               // current p2p version
+                Using<CustomUintFormatter<8>>(their_services),  // their service flags
+                int64_t{},                                      // dummy time
+                int64_t{},                                      // ignored service bits
+                CService{},                                     // dummy
+                int64_t{},                                      // ignored service bits
+                CService{},                                     // ignored
+                uint64_t{1},                                    // dummy nonce
+                std::string{},                                  // dummy subver
+                int32_t{},                                      // dummy starting_height
+                true),                                          // fRelay
+    };
+
+    (void)connman.ReceiveMsgFrom(node, msg_version);
+    node.fPauseSend = false;
+    connman.ProcessMessagesOnce(node);
+
+    // Receive messages
+    {
+        LOCK(node.cs_sendProcessing);
+        peerman.SendMessages(&node);
+    }
+
+    TestOnlyResetTimeData();
+
+    if (node.fDisconnect) return;
```

Comment:
c132abf1667ea9367ce797d808d6b6c885196601: Why would this disconnect? Seems like copy-pasted dead code?

Also, it looks like you forgot to copy-paste the peer-state asserts.

Also, I don't see why this is copied in the first place. In the commit message you claim that it is better to make the version handshake consistent with "live code". So wouldn't it be better to make this helper easily accessible for all tests instead of forcing reviewers to re-do the copy-paste every time it is and will be used?

See https://github.com/bitcoin/bitcoin/pull/25591, which should also make this pull's diff smaller.

Reply:
Rebased on top of the changes in #25591. This test now uses `ConnmanTestMsg::Handshake()`.